packages/react-components/react-toast/src/state/useToast.ts

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import { Toast } from './vanilla/toast';
+import { useIsomorphicLayoutEffect } from '@fluentui/react-utilities';
+
+export function useToast() {
+  const [toast] = React.useState(() => new Toast());
+  const [_, forceRender] = React.useReducer(() => ({}), {});
+  useIsomorphicLayoutEffect(() => {
+    toast.onUpdate = forceRender;
+  }, [toast]);
```

Comment:
```suggestion
const forceUpdate = useForceUpdate()
const [toast] = React.useState(() => {
  const newToast = new Toast()
  newToast.onUpdate = forceUpdate
  return newToast
});
```

I guess you could do it like this since this reference never changes?!

Reply:
Also, why not `useMemo` with no deps instead of `useState` with initial method?!

Reply:
> Also, why not useMemo with no deps instead of useState with initial method?!

Just habit really, there's always that scary warning in React docs that Memo is not a guarantee 🤷